columnflow/calibration/cms/egamma.py

```diff
@@ -300,52 +310,67 @@ def call_func(self, events: ak.Array, **kwargs) -> ak.Array:
             )
 
         # the correction tool only supports flat arrays, so convert inputs to flat np view first
-
         sceta = flat_np_view(events[self.source_field].superclusterEta, axis=1)
         r9 = flat_np_view(events[self.source_field].r9, axis=1)
         flat_seeds = flat_np_view(events[self.source_field].deterministic_seed, axis=1)
 
-        # prepare arguments
-        # we use pt as et since there depends in linear (following the recoomendations)
+        # we use pt as et since there depends in linear (following the recommendations)
         # (energy is part of the LorentzVector behavior)
+        pt = flat_np_view(events[self.source_field].rawPt, axis=1)
+
+        # prepare arguments
         variable_map = {
+            "AbsScEta": np.abs(sceta),
             "eta": sceta,
             "r9": r9,
+            "pt": pt,
             **self.resolution_cfg.corrector_kwargs,
         }
+
         args = tuple(
-            variable_map[inp.name] for inp in self.resolution_corrector.inputs
+            variable_map[inp.name]
+            for inp in self.resolution_corrector.inputs
             if inp.name in variable_map
         )
 
         # calculate the smearing scale
-        rho = self.resolution_corrector("rho", *args)
+        # available option are for Et: smear, esmear or escale, e = error
+
+        # TODO : ask Marcel/philip about this
+        # NOTE: the presence of the scale uncertainties ("escale") among the "EGMSmearAndSyst",
+        # as mentioned in the example above, allows us to apply them directly to the MC simulation.
+        rho = self.resolution_corrector.evaluate(self.resolution_cfg.corrector_strings["type"], *args)
 
         # -- stochastic smearing
         # normally distributed random numbers according to EGamma resolution
 
         # varied corrections
         if self.with_uncertainties and self.dataset_inst.is_mc:
-            rho_unc = self.resolution_corrector("err_rho", *args)
+            rho_unc = self.resolution_corrector(self.resolution_cfg.corrector_strings["unc_type"], *args)
+            random_normal_number = functools.partial(ak_random, 0, 1)
+            smearing_func = lambda rng_array, variation: rng_array * variation + 1
+
             smearing_up = (
-                ak_random(
-                    1, rho + rho_unc, flat_seeds,
-                    rand_func=self.deterministic_normal_up,
+                smearing_func(
+                    random_normal_number(flat_seeds, rand_func=self.deterministic_normal),
```

Comment:
I think this should be the dedicated random function for the up variation to avoid unwanted correlations between the random numbers for the nominal and shifted corrections:

```suggestion
                    random_normal_number(flat_seeds, rand_func=self.deterministic_normal_up),
```